### DISCUSSION TITLE
2022-09-07 Meeting Agenda, Recording & Notes

### DISCUSSION THREAD

**Discussion by andy-blum** [ddev/ddev#3963] [DDEV Advisory Group]
Previous meeting: #3800
Next meeting: #4191

## Agenda

- Turn on recording, ask for a minutes-taker
- Who will be at Drupalcon Prague or Build Anything Together?
- Discussion of code of conduct options, see [https://github.com/drud/ddev/issues/4159](https://github.com/drud/ddev/issues/4159)
- "Bus factor" updates, shared passwords, etc.
- Upstream donations status
- Financial status
- Anything on anybody's mind!

## Meeting Notes

**Attendees**
- Randy Fay
- Matt Stein
- Chris Wells
- Cary Gordon
- Andrew Berry
- Andrew Welch
- Andy Blum
- Benjamin Mack
- Nico (?)
- Robert Lang
- Jonas Eberle (late)
- Graham (late)

---

- _Started with Introductions._
  - f note: Matt Stein is here from CraftCMS - lots of new CraftCMS stuff because they've made it their preferred local dev env - this past weekend there was a large push for DDEV + CraftCMS
- _Upcoming events_ -
  - Andy Blum & Chris Wells (and Randy) at DrupalCon Prague
- _Lullabot is also looking at sponsoring DDEV financially._ One thing they want is a Code of Conduct. There's [an open PR for the GitHub developer code](https://github.com/drud/ddev/pull/4186).
- "_Bus Factor" discussion_:
  - Storage and retrieval of "secrets" - we have [an open issue and there is work](https://github.com/drud/ddev/pull/4175).
  - Need another account on the "team" at Apple so they have privileges for code signing, etc. The person needs the privilege, but _also_ to understand all the stuff that goes on there.
  - This leads to the need for a half-time co-maintainer. Someone who knows how to do a release, ex.g. Should be a Go developer (or be willing to become one). Andrew Welch created a [legal "Key Man" agreement](https://corporatefinanceinstitute.com/resources/knowledge/deals/key-man-clause/) in previous software shop that he was at; this may be an additional thing we need once we secure said "Key Person." Currently Simon Gilles is a backup person in GitHub. He has experience in every area of DDEV currently, so is the logical choice currently. But, he has a family and a job, etc… so…
  - Randy did make a budget proposal to Platform.sh (paraphrasing) "Can I get 20% of a doc person and 20% of a marketing person?" Not sure if this will gain traction. But it also gives him the idea that there are other agencies out there who might be able to sponsor a co-maintainer. This would give us an enormous amount of resiliency.
  - Commentary:
    - Good documentation is hard
    - Understanding this person as an apprentice or understudy is the right thought, because expecting someone to just jump right on may not be realistic.
    - What kinds of things can we carve off that are smaller? Because Redfin could give not 20 hrs/week, but maybe 4.
      - Andy B: Too many and it'll need a pm!
      - Product Owner / PM; Marketing & promotion; test runner maintenance; testing; DDEV get / plugin maintenance and oversight; docs & tech writing; community management & support
      - Requires someone in the small slots to be very interested in doing the things in that niche.
    - Andrew Welch - if we want to start this, start thinking about what are the nuts & bolts they need just to get the access. They don't need all the knowledge, but they need the tools that will let them figure it out on their own. (This is the intent with the secrets/keys issue: #3962)
    - A "Succession Plan" is what we need (rfay)
- _Upstream project donations_. Almost at the upstream 10%. We depend on hundreds of projects, so it's good to do that.
- _Financial updates._ About $8700 in the bank accounts. Moved the bank account to a more real business account at Wells Fargo. Seems fine there. There's a last bit in the old account that still needs to move. Wanted to make sure there were no more transactions going to the old account, and I think that's true now.
  - Could use some of this money to pay someone to do some "chunk" of work. Randy will look at how to spend this and how to do this. (Testing, etc? What will it be?)
  - We need to enhance the documentation and marketing in general just to make it known that you _can_ support DDEV via GitHub sponsors, etc. NOTE: "no tier is too high" -- maybe no one uses it, but hey some company may really just need DDEV and throws the money in there.
  - Will approach Pantheon at DrupalCon.
- _Other issues / thoughts?_
  - Cary - Should we have a known release cycle instead? Or should we slow down?
    - Randy - when people have issues, that generally prompts a release. And I hate seeing people have issues!
    - Cary - maybe don't release unless the docs are updated and people know they need to be on the latest. It's harder to support when older versions continue to work.
    - Nico - I actually like the fast pace of releases because I wouldn't be able to wait for a new release to get something, it would've blocked my work. In general, Andrew Welch & Chris Wells echo'd Nico's sentiment.

### Action Items

- [ ] Add account to "team" at apple
- [ ] Bring on half-time co-maintainer
- [ ] Begin documenting a "Succession Plan" (nice way to say "bus factor")
- Other things to document as part of this:
  - [ ] DDEV's goals
  - [ ] Release process (already documented but can't be sure it's accurate enough to make someone new be successful) (Simon _has_ done this)
  - [ ] Any tasks you do, they need to be written down if you intend to hand that off (ex.g. Running a full test suite against the newest version of docker-compose, Colima, Docker Desktop, etc.)
- [x] Set up [successor settings](https://github.com/settings/admin) in GitHub on rfay's account

## Recording

Recording is at https://www.dropbox.com/s/4i8xqn29uieh7rh/20220907_ddev_advisory_group.mp4?dl=0

## Financial Reports

[Localdev Foundation Profit and Loss 2022-09-07-14_41.pdf](https://github.com/drud/ddev/files/9508228/Localdev.Foundation.Profit.and.Loss.2022-09-07-14_41.pdf)
[Localdev Foundation Balance Sheet 2022-09-07-14_40.pdf](https://github.com/drud/ddev/files/9508229/Localdev.Foundation.Balance.Sheet.2022-09-07-14_40.pdf)

**Comment by chrisfromredfin** [comment]
Actual agenda is

* Turn on recording, ask for a minutes-taker
* Who will be at Drupalcon Prague or Build Anything Together?
* Discussion of code of conduct options, see [https://github.com/drud/ddev/issues/4159](https://www.google.com/url?q=https://github.com/drud/ddev/issues/4159&sa=D&source=calendar&ust=1662994942626296&usg=AOvVaw3F1kImPoFD67QxWOrryO_H)
* "Bus factor" updates, shared passwords, etc.
* Upstream donations status
* Financial status
* Anything on anybody's mind!

**Comment by chrisfromredfin** [comment]
Meeting notes / minutes: 
https://docs.google.com/document/d/1Fsqr6ncHzyO3sX7AJOK-kPDu8wA2EXkQo079snhuols/edit?usp=sharing
